samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/shared/impl/assessment/SecureDeliveryServiceImpl.java

```diff
@@ -348,6 +401,18 @@ private void handlePlugin( String secureDeliveryPlugin ) {
 	
 		try
 		{
+			// This is a built-in integration with no additional JAR file
+			if (secureDeliveryPlugin.equalsIgnoreCase("proctorio")) {
+				ApplicationContext appCtx = new AnnotationConfigApplicationContext(SecureDeliveryProctorio.class);
```

Comment:
samigo-services is a lib in shared and creating an application context in this way will have consequences.
For example components are managed by ComponentManager and webapps have their own init tied to the servlet lifecycle. ApplicationContexts are managed by these entities meaning they have a start and they have an end.
This application context is continually generated with every call here.

Reply:
Resolving this comment as this was the pattern established here long before this PR. In Spring 5 there is [method](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/context/support/GenericApplicationContext.html#registerBean-java.lang.Class-org.springframework.beans.factory.config.BeanDefinitionCustomizer...-) for registering beans with the application context. 

I think the large questions to ask here is if this design is mean to look for and add a service to an existing spring context (either Sakai Application Context or a Web Application Context) or is it simply wanting to inject Sakai services that are needed (use SpringBeanAutowiringSupport).